### PR TITLE
add faust lang

### PR DESCRIPTION
**Pull request by magnetophon**
This is my first forray into emacs and lisp.
Probably a lot wrong with this!

I'm using my own fork of faust-mode untill [this](https://github.com/rukano/emacs-faust-mode/pull/2) is merged.

### CODE REVIEW

File: layers/+lang/extra-langs/packages.el [hunk 1]
```diff
@@ -12,6 +15,9 @@
 (defun extra-langs/init-arduino-mode ()
   (use-package arduino-mode :defer t))
 
+(defun extra-langs/init-faust-mode ()
+  (use-package faust-mode :defer t :mode "\\.dsp\\'" :mode "\\.lib\\'"))
```
Comment (magnetophon) [inline layers/+lang/extra-langs/packages.el]: Probably ```.lib``` is also used for other filetypes.
Not sure how to fix though...

Comment (StreakyCobra) [inline layers/+lang/extra-langs/packages.el]: [Here](http://filext.com/file-extension/lib) is a list of `.lib` file types. The only concern I have is for the "common-lisp (library)", but I don't know if it is commonly used or not… maybe someone else knows?

Comment (StreakyCobra) [inline layers/+lang/extra-langs/packages.el]: Also, you can group the two `:mode` in one expression, like for instance [here](https://github.com/syl20bnr/spacemacs/blob/develop/layers/finance/packages.el#L28).

**Comment by StreakyCobra** [comment]
@syl20bnr I don't like the increase of github fetcher recently, it slow down every update. Here it's especially annoying because it's a multi-purpose layer that a lot of people may have. They will be slowed down to check update on a package they will probably never need.

Can we maybe have a custom variable to enable it? Or do we have another solution except asking to put it on melpa?

**Comment by TheBB** [comment]
Agree, if this can go in MELPA it would be best. Ping @magnetophon.

The third option is to make it a local package.

File: layers/+lang/extra-langs/packages.el (hunk 1 above)
Comment (magnetophon) [inline layers/+lang/extra-langs/packages.el]: grouping done.

**Comment by magnetophon** [comment]
OK, thanks.
I'll wait.

**Comment by syl20bnr** [comment]
@magnetophon tell us when this is ready to be merged, thank you!

**Comment by rukano** [comment]
I merged the pull request on the main repo.

**Comment by magnetophon** [comment]
@rukano Thanks for merging.

You mentioned you're not maintaining this anymore. Does that mean you won't put it in MELPA?
I don't think I'm the right maintainer for this as I'm not even  sure yet I want to keep using emacs.

What's the proper way forward?

**Comment by rukano** [comment]
I don't know.. I use emacs, but haven't been using faust for years... so... no idea.. :-/

**Comment by magnetophon** [comment]
@TheBB What did you mean by local package?
Is that an option maybe?

**Comment by StreakyCobra** [comment]
@magnetophon Copying the faust-mode source code directly into spacemacs, like this is already done for some packages: https://github.com/syl20bnr/spacemacs/tree/develop/layers/%2Blang/python/local

**Comment by TheBB** [comment]
Since @rukano said "feel free to fork and develop it further" I think it would be possible for @magnetophon to submit it to MELPA. Either his own repo or @rukano's (with the assumption that he can get commit rights there).

**Comment by magnetophon** [comment]
@TheBB OK, it's in melpa now.

**Comment by TheBB** [comment]
Great, congratulations! :+1:

**Comment by magnetophon** [comment]
@TheBB any particular reason you've reviewed but not merged this?

**Comment by TheBB** [comment]
I tag it as review to remind myself, next time I'm merging PRs, that it can be merged. I don't merge PRs all the time, and in particular right now I'm waiting for @syl20bnr to merge a rather significant one, so I'll wait a bit.

**Comment by magnetophon** [comment]
@TheBB OK, thanks!

**Comment by magnetophon** [comment]
@TheBB OK, thanks!

**Comment by TheBB** [comment]
Thanks! Cherry-picked in develop. You can safely delete your branch.

**Comment by magnetophon** [comment]
@TheBB Great!
I'll do PR's (if any) to develop from now on.

**Pull request closed without merging**